code/ui/generalhud/playerinfo/PlayerInfo.cs

```diff
@@ -2,30 +2,29 @@
 using Sandbox.UI;
 using Sandbox.UI.Construct;
 
+using TTTReborn.Items;
 using TTTReborn.Player;
 using TTTReborn.Roles;
 
 namespace TTTReborn.UI
 {
-    public class PlayerInfo : ObservablePanel
+    public class PlayerInfo : Panel
     {
-        public PlayerInfo() : base()
+        public PlayerInfo()
         {
             StyleSheet.Load("/ui/generalhud/playerinfo/PlayerInfo.scss");
 
             new RolePanel(this);
             new IndicatorsPanel(this);
-
-            AutoHideWithoutObservablePlayer = true;
```

Comment:
Did you reimplemented this?

Reply:
For the inventory selection, we don't need it, but for the player info we do. I'm open to suggestions on how you'd like to implement it.

For now I just did...
```
SetClass("hide", Local.Pawn is not TTTPlayer player || (player.IsSpectator && !player.IsSpectatingPlayer));
```

```
public bool IsSpectator
{
    // TODO: Change to be spectator class once FirstPersonSpectateCamera is added as well.
    get => (Camera is not FirstPersonCamera);
}
```

and then in that other PR I remember you adding a derived class for Spectating cameras, we can make use of that.

Reply:
okay great, that's enough for me :D